### ISSUE TITLE
Basic Auth Configuration results in ReferenceError

### ISSUE THREAD

**Issue by cinhtau** [searchkit/searchkit#1235]
## Basic Auth not working

Setup basic auth against Elasticsearch

```javascript
connection: {
    host: "https://localhost:9200",
    auth: {
        username: 'elastic',
        password: 'changeme'
    }
}
```

## Environment

Used this according to docs

```html
<head>
    <script src="https://cdn.jsdelivr.net/npm/@searchkit/instantsearch-client@latest"></script>
    <script src="https://cdn.jsdelivr.net/npm/instantsearch.js@4"></script>
    <script src="https://cdn.jsdelivr.net/npm/searchkit@latest"></script>
    <link rel="stylesheet" href="algolia.min.css"/>
</head>
```

Default Elasticsearch installation with free security + basic auth configured.

## Details

This results in the console to 

```text
index.ts:32 ReferenceError: Buffer is not defined
    at J.performNetworkRequest (Transporter.ts:22:17)
    at J.msearch (Transporter.ts:35:35)
    at W.performSearch (index.ts:29:48)
    at W.handleInstantSearchRequests (index.ts:62:34)
    at c.search (instantsearch-client@latest:7:1275)
    at t._search (instantsearch.js@4:2:126765)
    at t.searchOnlyWithDerivedHelpers (instantsearch.js@4:2:117734)
    at i.search (instantsearch.js@4:2:169918)
    at instantsearch.js@4:2:164866
    at instantsearch.js@4:2:10484
performSearch @ index.ts:32
instantsearch.js@4:2 Uncaught (in promise) TypeError: Cannot convert undefined or null to object
    at Function.keys (<anonymous>)
    at new gr (instantsearch.js@4:2:102374)
    at instantsearch.js@4:2:127256
    at Array.forEach (<anonymous>)
    at t._dispatchAlgoliaResponse (instantsearch.js@4:2:127162)
DevTools failed to load source map: Could not load content for https://cdn.jsdelivr.net/npm/instantsearch.production.min.js.map: HTTP error: status code 404, net::ERR_HTTP_RESPONSE_CODE_FAILURE
DevTools failed to load source map: Could not load content for https://cdn.jsdelivr.net/npm/@searchkit/index.global.js.map: HTTP error: status code 404, net::ERR_HTTP_RESPONSE_CODE_FAILURE
```

## Code To Reproduce Issue

Example Page

```html
<html>
<head>
    <script src="https://cdn.jsdelivr.net/npm/@searchkit/instantsearch-client@latest"></script>
    <script src="https://cdn.jsdelivr.net/npm/instantsearch.js@4"></script>
    <script src="https://cdn.jsdelivr.net/npm/searchkit@latest"></script>
    <link rel="stylesheet" href="algolia.min.css"/>
</head>
<body>
<div class="ais-InstantSearch">

    <div class="left-panel">
        <h2>Type</h2>
        <div id="type-list"></div>
    </div>
    <div class="right-panel">
        <div id="searchbox" class="ais-SearchBox"></div>
        <div id="hits"></div>
        <div id="pagination"></div>
    </div>
</div>

<script type="text/javascript">
    /* global instantsearch algoliasearch */
    const sk = new Searchkit({
        connection: {
            host: "https://localhost:9200",
            auth: {
                username: 'elastic',
                password: 'changeme'
            }
        },
        search_settings: {
            highlight_attributes: ["name"],
            snippet_attributes: ["description"],
            search_attributes: [{field: "name", weight: 3}, "description", "categories"],
            result_attributes: ["name", "description", "price", "categories"],
            facet_attributes: [{
                field: "categories.keyword",
                type: "string",
                attribute: "categories",
            }, {
                field: "price",
                type: "numeric",
                attribute: "price",
            }],
        }
    })

    const search = instantsearch({
        indexName: "products",
        searchClient: SearchkitInstantsearchClient(sk)
    });

    search.addWidgets([
        instantsearch.widgets.searchBox({
            container: "#searchbox"
        }),
        instantsearch.widgets.refinementList({
            container: "#type-list",
            attribute: "categories"
        }),
        instantsearch.widgets.hits({
            container: "#hits",
            templates: {
                item(hit, {html, components}) {
                    return html`
                        <h2>
                            ${hit.__hitIndex}:
                            ${components.Highlight({attribute: 'name', hit})}
                        </h2>
                        <p>${components.Snippet({attribute: 'description', hit})}</p>
                    `;
                }
            }
        }),
        instantsearch.widgets.pagination({
            container: "#pagination"
        })
    ]);

    search.start();
</script>

</body>
</html>
```

**Comment by joemcelroy** [comment]
Ouch! Got a fix, i just need to test it. Hopefully later on today will publish this fix. Thanks for reporting it!

**Comment by sleithart** [comment]
any updates on this? running into the same issue

**Comment by cinhtau** [comment]
In the mean time, you can fall back to API key or have no auth at all. For a production setup though, it might be not good enough.

**Comment by sleithart** [comment]
i'm passing in a basic auth header instead for now

**Comment by joemcelroy** [comment]
Sorry for delay, wanted to have a test app so can validate the fixes. Working as expected!

**Comment by joemcelroy** [comment]
released in 4.4.1
